File: 1979-FindGreatestCommonDivisorofArray/1979-FindGreatestCommonDivisorofArray.py
# Last updated: 7/18/2026, 4:38:50 PM
1class Solution(object):
2    def findGCD(self, nums):
3        min_num = nums[0]
4        max_num = nums[0]
5        
6        for num in nums:
7            if num < min_num:
8                min_num = num
9            if num > max_num:
10                max_num = num
11                
12        return self.gcd(max_num, min_num)
13
14    def gcd(self, a, b):
15        while b != 0:
16            temp = b
17            b = a % b
18            a = temp
19        return a
20

File: 1979-FindGreatestCommonDivisorofArray/1979-FindGreatestCommonDivisorofArray.java
// Last updated: 7/18/2026, 4:24:50 PM
1class Solution {
2    public int findGCD(int[] nums) {
3        int min=nums[0];
4        int max=nums[0];
5        for(int num:nums){
6            if(num<min){
7                min=num;
8            }
9            if(num>max){
10                max=num;
11            }
12        }
13        return Gcd(min,max);
14        
15    }
16    public int Gcd(int a,int b){
17        while(b!=0){
18            int temp=b;
19            b=a%b;
20            a=temp;
21        }
22        return a;
23    }
24}